bittide-instances/exe/post-vex-riscv-test/Main.hs

```diff
@@ -64,59 +62,34 @@ case_testGdbProgram = do
 
     withCreateProcess openOcdProc $ \_ _ (fromJust -> openOcdStdErr) _ -> do
       hSetBuffering openOcdStdErr LineBuffering
-      expectLine openOcdStdErr waitForHalt
+      expectLine Nothing openOcdStdErr waitForHalt
 
       -- XXX: Picocom doesn't immediately clean up after closing, because it
       --      spawns as a child of the shell (start.sh). We could use 'exec' to
       --      make sure the intermediate shell doesn't exist, but this causes
       --      the whole test program to exit with signal 15 (??????).
-      withCreateProcess picocomProc $ \maybePicocomStdIn maybePicocomStdOut maybePicocomStdErr _ -> do
+      withCreateProcess picocomProc $ \maybePicocomStdIn maybePicocomStdOut _ _ -> do
         let
           picocomStdIn = fromJust maybePicocomStdIn
           picocomStdOut = fromJust maybePicocomStdOut
 
-          -- Create function to log the output of the processes
-          loggingSequence = do
-            threadDelay 1_000_000 -- Wait 1 second for data loggers to catch up
-            putStrLn "Picocom stdout"
-            picocomOut <- readRemainingChars picocomStdOut
-            putStrLn picocomOut
-            case maybePicocomStdErr of
-              Nothing -> pure ()
-              Just h -> do
-                putStrLn "Picocom StdErr"
-                readRemainingChars h >>= putStrLn
```

Comment:
I see you are removing picocom stderr printing from the logging infra, do you think we don't need it?

Reply:
Mostly just wasn't sure how to integrate it into the new stuff. I would probably have to include another argument to the function, something like `onErr` which would run in the error branches. Not sure if that would be overcomplicating things, though.